core/src/test/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacadeTest.java

```diff
@@ -0,0 +1,482 @@
+                SurfaceHolderRequester.Callback callback = invocation.getArgument(0);
+                callback.onSurfaceHolderReady(surfaceHolder);
+                return null;
+            }
+        }).when(surfaceHolderRequester).requestSurfaceHolder(any(SurfaceHolderRequester.Callback.class));
+
+        facade.setSurfaceHolderRequester(surfaceHolderRequester);
+        facade.setOnPreparedListener(preparedListener);
+        facade.setOnSizeChangedListener(videoSizeChangedListener);
+        facade.setOnCompletionListener(completionListener);
+        facade.setOnErrorListener(errorListener);
+    }
+
+    @Ignore("We should forward to listeners / make logs testable / throw exception.")
+    @Test
+    public void givenNoBoundSurfaceHolderRequester_whenPreparing_thenLogsNotAttachedWarning() {
+
+    }
+
+    @Test
+    public void whenPreparing_thenRequestsAudioFocus() {
+        givenMediaPlayerIsPrepared();
+
+        verify(audioManager).requestAudioFocus(null, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN);
+    }
+
+    @Test
+    public void whenPreparing_thenDoesNotReleaseMediaPlayer() {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer, never()).reset();
+        verify(mediaPlayer, never()).release();
+    }
+
+    @Test
+    public void whenPreparingMultipleTimes_thenReleasesMediaPlayer() {
+        facade.prepareVideo(ANY_URI);
+        facade.prepareVideo(ANY_URI);
+
+        verify(mediaPlayer).reset();
+        verify(mediaPlayer).release();
+    }
+
+    @Test
+    public void whenPreparing_thenSetsDataSource() throws IOException {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer).setDataSource(context, ANY_URI, NO_HEADERS);
+    }
+
+    @Test
+    public void whenPreparing_thenSetsDisplay() {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer).setDisplay(surfaceHolder);
+    }
+
+    @Test
+    public void whenPreparing_thenSetsStreamMusicAudioStreamType() {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer).setAudioStreamType(AudioManager.STREAM_MUSIC);
+    }
+
+    @Test
+    public void whenPreparing_thenSetsScreenOnWhilePlayerToTrue() {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer).setScreenOnWhilePlaying(SCREEN_ON);
+    }
+
+    @Test
+    public void whenPreparing_thenPreparesMediaPlayerAsyncronously() {
+        givenMediaPlayerIsPrepared();
+
+        verify(mediaPlayer).prepareAsync();
+    }
+
+    @Ignore("We should forward to listeners / make logs testable / throw exception.")
+    @Test
+    public void givenException_whenPreparing_thenLogsCreationError() {
+
+    }
+
+    @Test
+    public void givenBoundPreparedListener_andMediaPlayerIsPrepared_whenPrepared_thenForwardsOnPrepared() {
+        facade.prepareVideo(ANY_URI);
+        ArgumentCaptor<MediaPlayer.OnPreparedListener> argumentCaptor = ArgumentCaptor.forClass(MediaPlayer.OnPreparedListener.class);
```

Comment:
We do. This is the explicit test though and I want to avoid hiding what is occurring inside of the method.